### PR TITLE
Actually optimise instrument list iteration

### PR DESCRIPTION
**Pull request by dsavchenko**
This was rather dumb mistake, so the idea of optimisation of instruments iteration didn't really work...

### CODE REVIEW

**Comment by volodymyrss** [comment]
Maybe we could add a timer to notice how long it takes?

**Review by volodymyrss — DISMISSED** [review]

**Comment by dsavchenko** [comment]
It shouldn't take too long with optimisation and will also decrease and stabilise when parameter lists are cached. But this logging makes sense, will add it. 

This problem arises only with nb2workflow plugin. So probably better (and easier, there is only one place) to add a timer there.
Does sentry of dispatcher accessible from plugins? I'm just not sure

**Review by burnout87 — APPROVED** [review]

**Review by volodymyrss — APPROVED** [review]

File: cdci_data_analysis/flask_app/dispatcher_query.py [hunk 1]
```diff
@@ -445,19 +446,24 @@ def get_user_specific_instrument_list(app):
...
-        for instrument_factory in importer.instrument_factory_iter:
-            if hasattr(instrument_factory, 'instrument_query'):
-                instrument_query = instrument_factory.instrument_query
-                instr_name = getattr(instrument_factory, 'instr_name', instrument_factory().name)
-            else:
-                instrument = instrument_factory()
-                instrument_query = instrument.instrumet_query
-                instr_name = instrument.name
-                
+        with block_timer(logger=logger,
```
Comment (burnout87) [inline cdci_data_analysis/flask_app/dispatcher_query.py:449]: no sentry?

**Review by burnout87 — APPROVED** [review]

File: cdci_data_analysis/flask_app/dispatcher_query.py (hunk 1 above)
Comment (dsavchenko) [reply cdci_data_analysis/flask_app/dispatcher_query.py:449]: It's [being set by default](https://github.com/oda-hub/dispatcher-app/blob/d841260bcb3b9208440244facc308d83f93e7180/cdci_data_analysis/timer.py#L7) to the dispatcher's main sentry instance

**Pull request merged**